### PR TITLE
Throw error on calling size() on a subroutine 

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fix https://github.com/lfortran/lfortran/issues/1899

### CODE REVIEW

File: tests/errors/subroutine3.f90 [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+    call f()
+    
+    contains
+    
+        subroutine f()
+        print *, size(f)
+        end subroutine
+    
+    end program
```
Comment (Pranavchiku) [inline tests/errors/subroutine3.f90:11]: Appreciate if you can indent this properly.

**Review by Pranavchiku — COMMENTED** [review]
I left a comment, rest all looks good to me. Thanks!

File: tests/errors/subroutine3.f90 (hunk 1 above)
Comment (HarshitaKalani) [reply tests/errors/subroutine3.f90]: Done, thanks.

File: src/lfortran/semantics/ast_common_visitor.h [hunk 2]
```diff
@@ -3566,6 +3566,15 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
             size_compiletime = ASRUtils::EXPR(ASR::make_IntegerConstant_t(al, x.base.base.loc,
                                                 compile_time_size, type));
         }
+        //if v_Var is a Function, give error
+        if(ASR::is_a<ASR::Var_t>(*v_Var)){
+            ASR::Var_t* var = ASR::down_cast<ASR::Var_t>(v_Var);
+            ASR::symbol_t* sym = var->m_v;
+            if(ASR::is_a<ASR::Function_t>(*sym)){
+                throw SemanticError("Argument of `size` must be an array", x.base.base.loc);
+            }
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:3575]: This should probably be made more general and reject everything except an expression of a type Array. But for now that's good.

**Review by certik — APPROVED** [review]
I think that looks good, thanks for fixing it!

**Pull request merged**